mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -1607,6 +1602,11 @@ LogicalResult CppEmitter::emitOperation(Operation &op, bool trailingSemicolon) {
        shouldBeInlined(cast<emitc::ExpressionOp>(op))))
     return success();
 
+  // Never emit semicolon for operations that end with } or opaque.
```

Comment:
We could update this to
```suggestion
  // Never emit a semicolon for some operations, especially if ending with `}`.
```
as also `cf::CondBranchOp` and `emitc::IncludeOp` don't end with `}`.